servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -128,6 +140,66 @@ private GetNamedRefsParams getGetNamedRefsParams(boolean fetchAdditionalInfo) {
         : GetNamedRefsParams.DEFAULT;
   }
 
+  /**
+   * Applies different filters to the {@link Stream} of references on the query expression.
+   *
+   * @param references The references that different filters will be applied to
+   * @param queryExpression The query expression to filter by
+   * @return A potentially filtered {@link Stream} of commits based on the query expression
+   */
+  private Stream<Reference> filterReferences(Stream<Reference> references, String queryExpression) {
+    if (Strings.isNullOrEmpty(queryExpression)) {
+      return references;
+    }
+
+    final Script script;
+    try {
+      script =
+          SCRIPT_HOST
+              .buildScript(queryExpression)
+              .withContainer(CONTAINER)
+              .withDeclarations(REFERENCES_DECLARATIONS)
+              .withTypes(REFERENCES_TYPES)
+              .build();
+    } catch (ScriptException e) {
+      throw new IllegalArgumentException(e);
+    }
+    return references.filter(
+        reference -> {
+          try {
+            ReferenceMetadata refMeta = reference.getMetadata();
+            if (refMeta == null) {
+              refMeta = CELUtil.EMPTY_REFERENCE_METADATA;
+            }
+            CommitMeta commit = refMeta.getCommitMetaOfHEAD();
+            if (commit == null) {
+              commit = CELUtil.EMPTY_COMMIT_META;
+            }
+            String refType;
+            if (reference instanceof Branch) {
+              refType = "Branch";
```

Comment:
I wonder if it might be preferable to use upper case `"BRANCH"` here to match the type name used in JSON payloads...

Reply:
Good point - changed